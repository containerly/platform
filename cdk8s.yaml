language: typescript
app: npx ts-node main.ts
imports:
  - https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.32.0/crds.yaml
