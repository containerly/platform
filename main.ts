import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';
import { OperatorGroup, Subscription } from './imports/operators.coreos.com';
import { Namespace } from 'cdk8s-plus-29';

export class Platform extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = { }) {
    super(scope, id, props);

    // Flux Namespace
    new Namespace(this, 'flux-system', {
      metadata: {
        name: 'flux-system',
      },
    });

    // Flux Operator Group
    new OperatorGroup(this, 'flux-operator-group', {
      metadata: {
        name: 'flux-operator-group',
        namespace: 'flux-system',
      },
      spec: {
        targetNamespaces: ['flux-system'],
      },
    });

    // Prometheus Operator Group
    new OperatorGroup(this, 'prometheus-operator-group', {
      metadata: {
        name: 'prometheus-operator-group',
        namespace: 'default',
      },
      spec: {
        targetNamespaces: ['default'],
      },
    });

    // Flux Subscription
    new Subscription(this, 'flux-subscription', {
      metadata: {
        name: 'flux-subscription',
        namespace: 'flux-system',
      },
      spec: {
        channel: 'stable',
        installPlanApproval: 'Automatic',
        name: 'flux',
        source: 'operatorhubio-catalog',
        sourceNamespace: 'olm',
      },
    });

    // Grafana Subscription
    new Subscription(this, 'grafana-subscription', {
      metadata: {
        name: 'grafana-subscription',
        namespace: 'operators',
      },
      spec: {
        channel: 'v5',
        name: 'grafana-operator',
        source: 'operatorhubio-catalog',
        sourceNamespace: 'olm',
      },
    });

    // Prometheus Subscription
    new Subscription(this, 'prometheus-subscription', {
      metadata: {
        name: 'prometheus-subscription',
        namespace: 'operators',
      },
      spec: {
        channel: 'beta',
        installPlanApproval: 'Automatic',
        name: 'prometheus',
        source: 'operatorhubio-catalog',
        sourceNamespace: 'olm',
      },
    });

    // Loki Subscription
    new Subscription(this, 'loki-subscription', {
      metadata: {
        name: 'loki-operator',
        namespace: 'operators',
      },
      spec: {
        channel: 'alpha',
        name: 'loki-operator',
        source: 'operatorhubio-catalog',
        sourceNamespace: 'olm',
      },
    });

    // Strimzi Kafka Subscription
    new Subscription(this, 'strimzi-kafka-subscription', {
      metadata: {
        name: 'strimzi-kafka-operator',
        namespace: 'operators',
      },
      spec: {
        channel: 'stable',
        name: 'strimzi-kafka-operator',
        source: 'operatorhubio-catalog',
        sourceNamespace: 'olm',
      },
    });

    // MinIO Subscription
    new Subscription(this, 'minio-subscription', {
      metadata: {
        name: 'minio-operator',
        namespace: 'operators',
      },
      spec: {
        channel: 'stable',
        name: 'minio-operator',
        source: 'operatorhubio-catalog',
        sourceNamespace: 'olm',
      },
    });

    // Elastic Cloud ECK Subscription
    new Subscription(this, 'elastic-cloud-eck-subscription', {
      metadata: {
        name: 'my-elastic-cloud-eck',
        namespace: 'operators',
      },
      spec: {
        channel: 'stable',
        name: 'elastic-cloud-eck',
        source: 'operatorhubio-catalog',
        sourceNamespace: 'olm',
      },
    });
  }
}

const app = new App();
new Platform(app, 'platform');
app.synth();
