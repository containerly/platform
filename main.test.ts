import {Platform} from './main';
import {Testing} from 'cdk8s';

describe('Platform', () => {
  test('Should synthesize correctly', () => {
    const app = Testing.app();
    const chart = new Platform(app, 'test-chart');
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
