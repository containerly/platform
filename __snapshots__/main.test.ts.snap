// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform Should synthesize correctly 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "name": "flux-system",
    },
    "spec": Object {},
  },
  Object {
    "apiVersion": "operators.coreos.com/v1",
    "kind": "OperatorGroup",
    "metadata": Object {
      "name": "flux-operator-group",
      "namespace": "flux-system",
    },
    "spec": Object {
      "targetNamespaces": Array [
        "flux-system",
      ],
    },
  },
  Object {
    "apiVersion": "operators.coreos.com/v1",
    "kind": "OperatorGroup",
    "metadata": Object {
      "name": "prometheus-operator-group",
      "namespace": "default",
    },
    "spec": Object {
      "targetNamespaces": Array [
        "default",
      ],
    },
  },
  Object {
    "apiVersion": "operators.coreos.com/v1alpha1",
    "kind": "Subscription",
    "metadata": Object {
      "name": "flux-subscription",
      "namespace": "flux-system",
    },
    "spec": Object {
      "channel": "stable",
      "installPlanApproval": "Automatic",
      "name": "flux",
      "source": "operatorhubio-catalog",
      "sourceNamespace": "olm",
    },
  },
  Object {
    "apiVersion": "operators.coreos.com/v1alpha1",
    "kind": "Subscription",
    "metadata": Object {
      "name": "grafana-subscription",
      "namespace": "operators",
    },
    "spec": Object {
      "channel": "v5",
      "name": "grafana-operator",
      "source": "operatorhubio-catalog",
      "sourceNamespace": "olm",
    },
  },
  Object {
    "apiVersion": "operators.coreos.com/v1alpha1",
    "kind": "Subscription",
    "metadata": Object {
      "name": "prometheus-subscription",
      "namespace": "operators",
    },
    "spec": Object {
      "channel": "beta",
      "installPlanApproval": "Automatic",
      "name": "prometheus",
      "source": "operatorhubio-catalog",
      "sourceNamespace": "olm",
    },
  },
  Object {
    "apiVersion": "operators.coreos.com/v1alpha1",
    "kind": "Subscription",
    "metadata": Object {
      "name": "loki-operator",
      "namespace": "operators",
    },
    "spec": Object {
      "channel": "alpha",
      "name": "loki-operator",
      "source": "operatorhubio-catalog",
      "sourceNamespace": "olm",
    },
  },
  Object {
    "apiVersion": "operators.coreos.com/v1alpha1",
    "kind": "Subscription",
    "metadata": Object {
      "name": "strimzi-kafka-operator",
      "namespace": "operators",
    },
    "spec": Object {
      "channel": "stable",
      "name": "strimzi-kafka-operator",
      "source": "operatorhubio-catalog",
      "sourceNamespace": "olm",
    },
  },
  Object {
    "apiVersion": "operators.coreos.com/v1alpha1",
    "kind": "Subscription",
    "metadata": Object {
      "name": "minio-operator",
      "namespace": "operators",
    },
    "spec": Object {
      "channel": "stable",
      "name": "minio-operator",
      "source": "operatorhubio-catalog",
      "sourceNamespace": "olm",
    },
  },
  Object {
    "apiVersion": "operators.coreos.com/v1alpha1",
    "kind": "Subscription",
    "metadata": Object {
      "name": "my-elastic-cloud-eck",
      "namespace": "operators",
    },
    "spec": Object {
      "channel": "stable",
      "name": "elastic-cloud-eck",
      "source": "operatorhubio-catalog",
      "sourceNamespace": "olm",
    },
  },
]
`;
